### ISSUE TITLE
beginWith() is no longer doing prefix matching in Objective-C

### ISSUE THREAD

**Issue by jspahrsummers** [Quick/Nimble#76]
Example failure messages from Nimble 0.2:

```
/var/lib/jenkins/workspace/Mac/UnitTests/GHCommitMessageViewModelSpec.m:94: failed - expected to begin with <Conflicts:>, got <Conflicts:README>:
91 		waitUntilCompleted:NULL];
92 
93 	expect(viewModel.commitSummary).withTimeout(5).toEventually(equal(@"Merge branch 'new'"));
94 	expect(viewModel.commitDescription).to(beginWith(@"Conflicts:"));
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
95 });
96 
```

**Comment by jspahrsummers** [comment]
This was from the [v0.2.0](https://github.com/Quick/Nimble/releases/tag/v0.2.0) tag, and doesn't occur on 56c3b79 (which was the version we were using before). I'm afraid I didn't investigate much further than this, as I was just trying to get a buildable project—I ended up just reverting to that commit of Nimble.

That property is an `NSString` (in Objective-C, though), and I'm using Xcode 6.1.1, build version 6A2006.

**Comment by rbeeger** [comment]
I tried to reproduce this without any success.

I used:
  * XCode 6.1.1 (6A2008a) with Nimble's swift-1.1 branch
  * XCode 6.3 (6D250o) with Nimble's master branch

Does this bug still occur for you @jspahrsummers or has it disappeared because of updates to XCode or Nimble in the meantime?

**Comment by jspahrsummers** [comment]
This appears to be fixed as of v0.3.0 (or perhaps earlier). :sparkles:

**Comment by modocache** [comment]
Awesome, thanks for the diagnoses and follow-up, @rbeeger and @jspahrsummers! :+1:
